File: ShipClass/src/ship.cpp
//FILE HEADER

#include "ship.h"

float
toRadians(float degrees)
{
  return (PI * degrees) / 180;
}

///////////////////////////////////////////////////////

Ship::Ship()
{
  location = {0, 0};
  velocity = {0, 0};
  angleDeg = 0;
  radius = 5;
  maxLocation = {1000, 1000};
}

//=======================================================

void
Ship::setLocation(float x, float y)
{
  //validate
  location.x = x;
  location.y = y;
}

void
Ship::setVelocity(float velX, float velY)
{
  velocity.x = velX;
  velocity.y = velY;
}

void
Ship::setAngle(float angle)
{
  //validate
  angleDeg = angle;
}

//========================================================

void
Ship::turnLeft()
{
  angleDeg++;
  if (angleDeg == 360)
    angleDeg = 0;
}

void
Ship::turnRight()
{
  angleDeg--;
  if (angleDeg == -1)
    angleDeg = 359;
}

void
Ship::applyThrust(int thrust)
{
  velocity.y = std::sin(toRadians(angleDeg)) * thrust;
  velocity.x = std::cos(toRadians(angleDeg)) * thrust;
  updateLocation();
}

void
Ship::updateLocation()
{
  location.x = location.x + velocity.x;
  location.y = location.y + velocity.y;
}

//========================================================

float
Ship::getRadius()
{
  return radius;
}

MyVect
Ship::getLocation()
{
  return location;
}

MyVect
Ship::getVelocity()
{
  return velocity;
}

float
Ship::getAngle()
{
  return angleDeg;
}

File: ShipTestPy/src/Ship.py
import math    

#No vector class created, it does not make as much since
#in this case, because a list class already exists
def listToFloat(data):
    for i in range(len(data)):
        data[i] = float(data[i])

class Ship:
    #Python does not have a private declaration,
    #but it does have something called name mangling
    #that will change the names of the members to include
    #the class name so that they can't be called directly
    __maxLocation = [1000, 1000]
    __location = [500, 500]
    __velocity = [0, 0]
    __angleDeg = 0
    __radius = 5
    
    #no "default" constructor necessary, values already
    #initialized
    ########################################################
    def setLocation(self, new):
        self.__location[0] = new[0]
        self.__location[1] = new[1]
        
    def setVelocity(self, new):
        self.__velocity[0] = new[0]
        self.__velocity[1] = new[1]
        
    def setAngle(self, angle):
        self.__angleDeg = angle
    ########################################################
    def updateLocation(self):
        self.__location[0] += self.__velocity[0]
        self.__location[1] += self.__velocity[1]
    
    def turnLeft(self):
        self.__angleDeg += 1
        if self.__angleDeg == 360:
            self.__angleDeg = 0
    
    def turnRight(self):
        self.__angleDeg -= 1
        if self.__angleDeg == -1:
            self.angleDeg = 359
            
    def applyThrust(self, thrust):
        self.__velocity[0] = math.cos(math.radians(self.__angleDeg)) * thrust
        self.__velocity[1] = math.sin(math.radians(self.__angleDeg)) * thrust
        self.updateLocation()
    ############################################################
    def getRadius(self):
        return self.__radius
    def getLocation(self):
        return self.__location
    def getVelocity(self):
        return self.__velocity
    def getAngle(self):
        return self.__angleDeg

def output(ship):
    loc = ship.getLocation()
    vel = ship.getVelocity()
    ang = ship.getAngle()
    print('(', loc[0], ',', loc[1], '); <',
          vel[0], ',', vel[1], '>; ', ang, sep='')
